AppCenterDataStorage/AppCenterDataStorage/Internal/Client/MSTokenExchange.m

```diff
@@ -143,45 +143,28 @@ + (void)saveToken:(MSTokenResult *)tokenResult {
   }
 }
 
-/*
- * Get back the Cached Cosmos DB token
- * If the DB token has expired, that token is deleted from KeyChain
- */
-+ (MSTokenResult *_Nullable)retrieveCachedToken:(NSString *)partitionName {
-  if (partitionName) {
-    NSString *tokenString = [MSKeychainUtil stringForKey:[MSTokenExchange tokenKeyNameForPartition:partitionName]];
++ (MSTokenResult *_Nullable)retrieveCachedTokenForPartition:(NSString *)partition includeExpiredToken:(BOOL)includeExpiredToken {
+  if (partition) {
+    NSString *tokenString = [MSKeychainUtil stringForKey:[MSTokenExchange tokenKeyNameForPartition:partition]];
     if (tokenString) {
       MSTokenResult *tokenResult = [[MSTokenResult alloc] initWithString:tokenString];
       NSDate *currentUTCDate = [NSDate date];
       NSDate *tokenExpireDate = [MSUtility dateFromISO8601:tokenResult.expiresOn];
       if ([currentUTCDate laterDate:tokenExpireDate] == currentUTCDate) {
-        [MSTokenExchange removeCachedToken:partitionName];
```

Comment:
So the new behavior is to *never* remove the token and return it if they set the parameter?  So now we have to use a different method if we need to remove the token?

Reply:
The token should be persisted unless the user logout, so we never need to delete the token when user logged in :)